src/aiida/engine/daemon/execmanager.py

```diff
@@ -178,11 +173,11 @@ def upload_calculation(
             # Note: this will possibly overwrite files
             for root, dirnames, filenames in code.base.repository.walk():
                 # mkdir of root
-                transport.makedirs(str(root), ignore_existing=True)
+                transport.makedirs(workdir.join(root), ignore_existing=True)
 
                 # remotely mkdir first
                 for dirname in dirnames:
-                    transport.makedirs(str(root / dirname), ignore_existing=True)
+                    transport.makedirs(workdir.join(root, dirname), ignore_existing=True)
```

Comment:
Also here not sure what's wrong with the previous implementation?

Reply:
`root` could be a relative path to `workdir`